test/versioned/elastic/elasticsearch.tap.js

```diff
@@ -0,0 +1,323 @@
+    await helper.runInTransaction(agent, async function transactionInScope() {
+      t.ok(agent.getTransaction(), 'transaction should be visible')
+
+      try {
+        await client.bulk({
+          refresh: true,
+          operations: [
+            { index: { _index: DB_INDEX } },
+            { title: 'First Bulk Doc', body: 'Content of first bulk document' },
+            { index: { _index: DB_INDEX } },
+            { title: 'Second Bulk Doc', body: 'Content of second bulk document.' },
+            { index: { _index: DB_INDEX } },
+            { title: 'Third Bulk Doc', body: 'Content of third bulk document.' },
+            { index: { _index: DB_INDEX } },
+            { title: 'Fourth Bulk Doc', body: 'Content of fourth bulk document.' },
+            { index: { _index: DB_INDEX_2 } },
+            { title: 'Fifth Bulk Doc', body: 'Content of fifth bulk document' },
+            { index: { _index: DB_INDEX_2 } },
+            { title: 'Sixth Bulk Doc', body: 'Content of sixth bulk document.' },
+            { index: { _index: DB_INDEX_2 } },
+            { title: 'Seventh Bulk Doc', body: 'Content of seventh bulk document.' },
+            { index: { _index: DB_INDEX_2 } },
+            { title: 'Eighth Bulk Doc', body: 'Content of eighth bulk document.' }
+          ]
+        })
+      } catch (e) {
+        t.notOk(e, 'Bulk operations should not error')
+      }
+      const transaction = agent.getTransaction()
+      t.ok(transaction, 'transaction should still be visible after bulk create')
+      const trace = transaction.trace
+      t.ok(trace?.root?.children?.[0], 'trace, trace root, and first child should exist')
+      const firstChild = trace.root.children[0]
+      t.equal(
+        firstChild.name,
+        'Datastore/statement/ElasticSearch/other/_bulk',
+        'should record bulk operation'
+      )
+    })
+  })
+
+  t.test('should record search with query string', async function (t) {
+    await helper.runInTransaction(agent, async function transactionInScope() {
+      const search = await client.search({ index: DB_INDEX_2, q: 'sixth' })
+      t.ok(search, 'search should return a result')
+      const transaction = agent.getTransaction()
+      t.ok(transaction, 'transaction should still be visible after search')
+      const trace = transaction.trace
+      t.ok(trace?.root?.children?.[0], 'trace, trace root, and first child should exist')
+      const firstChild = trace.root.children[0]
+      t.match(
+        firstChild.name,
+        'Datastore/statement/ElasticSearch/test2/GET',
+        'querystring search should be recorded as a GET'
+      )
+      const attrs = firstChild.getAttributes()
+      t.match(attrs.product, 'ElasticSearch')
+      t.match(attrs.host, METRIC_HOST_NAME)
+      transaction.end()
+      // can we inspect recorded query?
+      agent.getTransaction().end()
+      t.ok(agent.queries.samples.size > 0, 'there should be a query sample')
```

Comment:
these are failing because you need more config to capture queries, take a look at the prisma versioned tests.

Reply:
I've updated those.